analyzers/tests/SonarAnalyzer.UnitTest/TestCases/DebuggerDisplayUsesExistingMembers.vb

```diff
@@ -0,0 +1,218 @@
+    Const ConstantFragment1 As String = "{Non"
+    Const ConstantFragment2 As String = "Existing}"
+
+    Property SomeProperty As Integer
+    Public SomeField As Integer
+
+    <DebuggerDisplayAttribute("1")> Property WithSuffix As Integer
+    <System.Diagnostics.DebuggerDisplay("1")> Property WithNamespace As Integer
+
+    <DebuggerDisplay(Nothing)> Property WithEmptyArgList As Integer
+    <DebuggerDisplay("")> Property WithEmptyFormat As Integer
+    <DebuggerDisplay(ConstantWithoutInvalidMembers)> Property WithFormatAsConstant1 As Integer
+    <DebuggerDisplay(NameOf(ConstantWithoutInvalidMembers))> Property WithFormatAsNameOf As Integer
+
+    <DebuggerDisplay("{SomeProperty}")> Property WithExistingProperty As Integer
+    <DebuggerDisplay("{SomeField}")> Property WithExistingField As Integer
+    <DebuggerDisplay("{SomeField}")> Property WithExistingFieldVerbatim As Integer
+    <DebuggerDisplay("{1 + 1}")> Property WithNoMemberReferenced1 As Integer
+    <DebuggerDisplay("{""1"" + ""1""}")> Property WithNoMemberReferenced2 As Integer
+
+    <DebuggerDisplay("{Nonexistent}")> Property WithNonexistentMember1 As Integer                          ' Noncompliant {{'Nonexistent' doesn't exist in this context.}}
+    '                ^^^^^^^^^^^^^^^
+    <DebuggerDisplay("1 + {Nonexistent}")> Property WithNonexistentMember2 As Integer                      ' Noncompliant {{'Nonexistent' doesn't exist in this context.}}
+    '                ^^^^^^^^^^^^^^^^^^^
+    <DebuggerDisplay("{Nonexistent1} bla bla {Nonexistent2}")> Property WithMultipleNonexistent As Integer ' Noncompliant {{'Nonexistent1' doesn't exist in this context.}}
+    '                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    <DebuggerDisplay("{Nonexistent}")> Property WithNonexistentMemberVerbatim As Integer                   ' Noncompliant {{'Nonexistent' doesn't exist in this context.}}
+    '                ^^^^^^^^^^^^^^^
+
+    <DebuggerDisplay(ConstantWithInvalidMember)> Property WithFormatAsConstant2 As Integer                            ' FN: constants are not checked
+    <DebuggerDisplay("{Non" & "Existing}")> Property WithFormatAsConcatenationOfLiterals As Integer                   ' FN: only simple literal supported
+    <DebuggerDisplay(ConstantFragment1 & ConstantFragment2)> Property WithFormatAsConcatenationOfConstants As Integer ' FN: only simple literal supported
+
+    <DebuggerDisplay("{Me.NonexistentProperty}")> Property PropertyWithExplicitThis As Integer                        ' FN: "Me." not supported (valid when debugging a VB.NET project)
+    <DebuggerDisplay("{this.NonexistentField}")> Property FieldWithExplicitThis As Integer                            ' FN: "this." not supported (valid when debugging a C# project)
+    <DebuggerDisplay("{1 + NonexistentProperty}")> Property ContainingInvalidMembers As Integer                       ' FN: expressions not supported
+End Class
+
+<DebuggerDisplay("{this.ToString()}")>
```

Comment:
Kept both: the language evaluated in braces is the one of the stack frame, and not the language in which the attribute is expressed.

c.p. https://github.com/SonarSource/sonar-dotnet/pull/6728#discussion_r1100410035

Reply:
I suggest to add a small comment in the test case when VB .NET in inside the C# file and vice-versa - otherwise it will be confusing for future readers

Reply:
Done.